### PR TITLE
GRAPHICS/GUI: Fix various undefined behaviours

### PR DESCRIPTION
**Pull request by lephilousophe**
These errors were detected with UBSanitizer

### CODE REVIEW

File: graphics/cursorman.cpp [hunk 1]
```diff
@@ -248,7 +248,8 @@ CursorManager::Cursor::Cursor(const void *data, uint w, uint h, int hotspotX, in
 	 else
 		_format = *format;
 	_size = w * h * _format.bytesPerPixel;
-	_keycolor = keycolor & ((1 << (_format.bytesPerPixel << 3)) - 1);
+	const uint32 keycolor_mask = (((uint32)-1) >> (sizeof(uint32)*8 - _format.bytesPerPixel * 8));
```
Comment (aquadran) [inline graphics/cursorman.cpp:251]: formatting

Comment (lephilousophe) [reply graphics/cursorman.cpp]: oops, should be better now

**Review by bluegr — APPROVED** [review]

**Comment by bluegr** [comment]
Nice work!

**Comment by bluegr** [comment]
Thanks! Merging

**Pull request merged**